### PR TITLE
SLO CORE - Added back Request verification

### PR DESCRIPTION
**Pull request by blen-desta**
#### What does this PR do (if it's not clear from the Title)? Please include any specification references that apply.
Added back Request verifications

#### Checklist:
- [x] SAML Spec Table of Contents documentation updated
- [ ] Unit Tests Added/Modified

### CODE REVIEW

**Comment by bakejeyner** [comment]
✋ I got the in-depth review on this one

**Comment by brjeter** [comment]
✋ I got the super-shallow review on this one.

**Comment by blen-desta** [comment]
@brjeter 
- Correct. IdP will be sending logout Requests to the second SP. 
- I can do that. It won't be much

File: ctk/common/src/main/kotlin/org/codice/compliance/verification/core/RequestVerifier.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import org.codice.compliance.SAMLCore_3_2_1_a
+import org.codice.compliance.SAMLCore_3_2_1_b
+import org.codice.compliance.SAMLCore_3_2_1_c
+import org.codice.compliance.attributeNode
+import org.codice.compliance.utils.TestCommon.Companion.ID
+import org.codice.compliance.utils.TestCommon.Companion.VERSION
+import org.w3c.dom.Node
+
+abstract class RequestVerifier(private val samlRequestDom: Node) : CoreVerifier(samlRequestDom) {
```
[the remark this answers is not in the document]
Comment (blen-desta) [reply ctk/common/src/main/kotlin/org/codice/compliance/verification/core/RequestVerifier.kt]: That's a negative test on the IdP. Not a verification.

File: ctk/idp/coverage/Core.md [hunk 2]
```diff
@@ -62,9 +62,9 @@ Unmarked sections need attention
 -			2.7.4.3 Element <Evidence>
 3 SAML Protocols
 -	3.1 Schema Header and Namespace Declarations
-	3.2 Requests and Responses
++	3.2 Requests and Responses
 +		3.2.1 Complex Type RequestAbstractType
-		3.2.2 Complex Type StatusResponseType
++		3.2.2 Complex Type StatusResponseType
```
[the remark this answers is not in the document]
Comment (blen-desta) [reply ctk/idp/coverage/Core.md]: Whoever completed that didn't update the coverage

File: ctk/common/src/main/kotlin/org/codice/compliance/verification/core/RequestVerifier.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (bakejeyner) [reply ctk/common/src/main/kotlin/org/codice/compliance/verification/core/RequestVerifier.kt]: But the IdP should be sending us a `<LogoutRequest>` in SLO. That means we need to verify that MUST.

[the remark this answers is not in the document]
Comment (blen-desta) [reply ctk/common/src/main/kotlin/org/codice/compliance/verification/core/RequestVerifier.kt]: Correct but what that is saying is the recipient of the LogoutRequest should verify the URI and not trust the request if it doesn't check out.
Which means when IdP sends the request, SP should check destination and when the SP (our test) send the request the IdP should check the destination. We will verify that the Idp is doing that by sending an incorrect destination and expecting an error.

**Comment by blen-desta** [comment]
build now

File: ctk/common/src/main/kotlin/org/codice/compliance/verification/core/RequestVerifier.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (brjeter) [reply ctk/common/src/main/kotlin/org/codice/compliance/verification/core/RequestVerifier.kt]: I think what @bakejeyner is saying is that the implied MUST here is that the IdP needs to send us a URI reference that points to the actual samlconf endpoint that we received it at. Which would be a positive test.

[the remark this answers is not in the document]
Comment (blen-desta) [reply ctk/common/src/main/kotlin/org/codice/compliance/verification/core/RequestVerifier.kt]: Yeah after discussing it further we decided to leave a card and wait to see how the second SP will be implemented

**Pull request merged**